bridge/slack/helpers.go

```diff
@@ -61,22 +60,16 @@ func (b *Bslack) getChannelByID(ID string) (*slack.Channel, error) {
 
 const minimumRefreshInterval = 10 * time.Second
 
-var (
-	refreshMutex           sync.Mutex
-	refreshInProgress      bool
-	earliestChannelRefresh = time.Now()
-	earliestUserRefresh    = time.Now()
-)
-
 func (b *Bslack) populateUsers() {
-	refreshMutex.Lock()
-	if time.Now().Before(earliestUserRefresh) || refreshInProgress {
-		b.Log.Debugf("Not refreshing user list as it was done less than %d seconds ago.", int(minimumRefreshInterval.Seconds()))
-		refreshMutex.Unlock()
+	b.refreshMutex.Lock()
+	defer b.refreshMutex.Unlock()
```

Comment:
We can't use a `defer` here. The purpose of manually unlocking it early, after the refresh timer and in-progress check is to enable other calls to this function not to block on it.